core/src/main/java/nl/nn/adapterframework/webcontrol/api/ShowConfigurationStatus.java

```diff
@@ -85,9 +86,20 @@ public Response getAdapters(@QueryParam("expanded") String expanded, @QueryParam
 	@RolesAllowed({"IbisObserver", "IbisDataAdmin", "IbisAdmin", "IbisTester"})
 	@Path("/adapters/{name}")
 	@Produces(MediaType.APPLICATION_JSON)
-	public Response getAdapter(@PathParam("name") String name, @QueryParam("configuration") String configuration, @QueryParam("expanded") String expanded, @QueryParam("showPendingMsgCount") boolean showPendingMsgCount) {
+	public Response getAdapterOld(@PathParam("name") String name, @QueryParam("configuration") String configuration, @QueryParam("expanded") String expanded, @QueryParam("showPendingMsgCount") boolean showPendingMsgCount) {
+		if(StringUtils.isNotEmpty(configuration)) {
+			return getAdapterNew(configuration, name, expanded, showPendingMsgCount);
+		}
+		throw new ApiException("either provide the configuration as query param or use endpoint /configurations/<config>/adapters/"+name, Status.MOVED_PERMANENTLY);
```

Comment:
Wat misschien 'aardiger' is, is om alleen een exceptie te gooien als er meer dan 1 adapter is met de gevraagde naam.

Reply:
Misschien wel aardiger, maar dat moet de API afweten van alle adapters en dat weet hij niet.